src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -274,8 +274,8 @@ function mpas_geotile_mgr_init(mgr, path) result(ierr)
         ! Calculate the total number of pixels in x dir
         ! NOTE: This calculation assumes that a dataset is a global dataset and may
         ! not work correctly for non-global datasets
-        mgr % pixel_nx = nint(360.0_RKIND / dx)
-        mgr % pixel_ny = nint(180.0_RKIND / dy)
+        mgr % pixel_nx = abs(nint(360.0_RKIND / dx))
+        mgr % pixel_ny = abs(nint(180.0_RKIND / dy))
```

Comment:
Just for consideration, an alternative here would be to call the `abs` intrinsic with only `dx` and `dy` as arguments, since the literal constants `360.0_RKIND` and `180.0_RKIND` can obviously never be negative. The result is the same, but it seems like it better expresses to the reader the possibility that `dx` and `dy` might be negative if these variables alone are the arguments to the `abs` call.

Reply:
I like this a lot better. The more succinct and clearer the better. I'll update it.

Reply:
Fixed.